src/renderer/entities/wallet/lib/import-keys-utils.ts

```diff
@@ -0,0 +1,138 @@
+import { ImportedDerivation, ImportFileChain, ImportFileKey, ParsedImportFile, TypedImportedDerivation } from './types';
+import { AccountId, ChainId, KeyType } from '@renderer/shared/core';
+import { chainsService } from '@renderer/entities/network';
+
+const IMPORT_FILE_VERSION = '1';
+
+export const importKeysUtils = {
+  isFileStructureValid,
+  getDerivationsFromFile,
+  isDerivationValid,
+  mergeChainDerivations,
+};
+
+function isFileStructureValid(result: any): result is ParsedImportFile {
+  const isVersionValid = 'version' in result && result.version === IMPORT_FILE_VERSION;
+  const hasPublicKey = Object.keys(result).every((key) => key.startsWith('0x') || key === 'version');
+
+  const genesisHashes = Object.values(result).filter((x) => typeof x === 'object') as ImportFileChain[];
+  const hasGenesisHash = Object.values(genesisHashes).every((hash) =>
+    Object.keys(hash).every((key) => key.startsWith('0x')),
+  );
+  const hasKeys = Object.values(genesisHashes).every((hash: ImportFileChain) =>
+    Object.values(hash).every(
+      (v) => Array.isArray(v) && v.every((keyObj) => 'key' in keyObj && Array.isArray(keyObj.key)),
+    ),
+  );
+
+  return isVersionValid && hasPublicKey && hasGenesisHash && hasKeys;
+}
+
+type FormattedResult = { derivations: ImportedDerivation[]; root: AccountId };
+function getDerivationsFromFile(result: ParsedImportFile): FormattedResult | undefined {
```

Comment:
`getDerivationsFromFile(result: ParsedImportFile)` - `result` is a misleading name for input parameter